cps/main.c

```diff
@@ -137,20 +139,66 @@ process_ingress(struct gatekeeper_if *iface, struct rte_kni *kni,
 }
 
 static void
-process_egress(struct gatekeeper_if *iface, struct rte_kni *kni,
-	uint16_t tx_queue)
+process_egress(struct cps_config *cps_conf, struct gatekeeper_if *iface,
+	struct rte_kni *kni, uint16_t tx_queue)
 {
 	struct rte_mbuf *bufs[GATEKEEPER_MAX_PKT_BURST];
+	struct rte_mbuf *cps_bufs[GATEKEEPER_MAX_PKT_BURST];
 	uint16_t num_rx = rte_kni_rx_burst(kni, bufs, GATEKEEPER_MAX_PKT_BURST);
-	unsigned int num_tx = rte_eth_tx_burst(iface->id, tx_queue,
-		bufs, num_rx);
+	struct acl_search acl = {
+		.num = 0,
+	};
+	uint16_t num_cps = 0;
+	unsigned int num_tx;
+	unsigned int i;
+	int ret;
 
-	/*
-	 * TODO Forward BGP, respond to ARP or ND requests with resolution,
-	 * and drop everything else.
-	 */
+	if (num_rx == 0)
+		return;
 
-	if (unlikely(num_tx < num_rx)) {
+	for (i = 0; i < num_rx; i++) {
+		acl.data[acl.num] = rte_pktmbuf_mtod(bufs[i], uint8_t *);
+		acl.mbufs[acl.num] = bufs[i];
+		acl.num++;
+	}
+
+	ret = rte_acl_classify(cps_conf->kni_acl, acl.data,
```

Comment:
Hi Cody,

Once we add support to extension headers of IPv6, the ACL will be so watered down here that it's not worth. Moreover, different from the global IPv6 filter that has more than a user, the only user of this ACL is the CPS itself. Please drop the ACL and just code directly.